pkg/proxy/iptables/proxier.go

```diff
@@ -332,12 +339,15 @@ func NewProxier(ipt utiliptables.Interface,
 		serviceChanges:           proxy.NewServiceChangeTracker(newServiceInfo, &isIPv6, recorder),
 		endpointsMap:             make(proxy.EndpointsMap),
 		endpointsChanges:         proxy.NewEndpointChangeTracker(hostname, newEndpointInfo, &isIPv6, recorder),
+		nodeMap:                  make(proxy.NodeMap),
+		nodeChanges:              proxy.NewNodeChangeTracker(),
 		iptables:                 ipt,
 		masqueradeAll:            masqueradeAll,
 		masqueradeMark:           masqueradeMark,
 		exec:                     exec,
 		clusterCIDR:              clusterCIDR,
 		hostname:                 hostname,
+		nodeName:                 types.NodeName(hostname),
```

Comment:
The entire point of the existence of `types.NodeName` is to avoid conflating the node name with the hostname...

Reply:
Kubelet use hostname as nodeName: https://github.com/kubernetes/kubernetes/blob/117e83157bc340c1823f5bd34d4f73ca126ccdd6/pkg/kubelet/kubelet.go#L388

Override nodeName only if uses cloud provider and there is a nodeName available in `CurrentNodeName()` :
https://github.com/kubernetes/kubernetes/blob/117e83157bc340c1823f5bd34d4f73ca126ccdd6/pkg/kubelet/kubelet.go#L396

But kube-proxy can not call cloud provider, any better idea ?